src/python/gem5/components/boards/kernel_disk_workload.py

```diff
@@ -187,6 +188,9 @@ def set_kernel_disk_workload(
         # full-system simulation.
         self._set_fullsystem(True)
 
+        if self.workload.object_file:
```

Comment:
You might be able to get access to `is_workload_set()` function if it is moved to `AbstractCore` as the `AbstractBoard` has an `AbstractProcessor` which has the `AbstractCore`, so it might be possible to use the same method to do this check as done in the `set_se_binary_workload` function.